openhtf/plugs/__init__.py

```diff
@@ -184,19 +185,29 @@ class PlugManager(object):
   main framework thread anyway.
 
   Attributes:
-    plug_map: Dict mapping plug type to instantiated plug.
+    _plug_map: Dict mapping plug type to instantiated plug.
   """
 
-  def __init__(self, plug_types):
-    self.plug_map = {}
+  def InitializePlugs(self, plug_types):
+    self._plug_map = {}
     try:
       for plug_type in plug_types:
-        self.plug_map[plug_type] = plug_type()
+        self._plug_map[plug_type] = plug_type()
     except Exception:  # pylint: disable=broad-except
       _LOG.exception('Exception insantiating plug type %s', plug_type)
       self.TearDownPlugs()
       raise
 
+  @property
+  def _asdict(self):
```

Comment:
ah, okay, i see what you did, you're relying on the recursive nature of `ConvertToBaseTypes`.. that's fine actually, but can you look into where that's actually called, and potentially just remove both `_asdict()` methods if they're not used?  they might be used for the http_api, I think...